tests/test_channels.py

```diff
@@ -345,3 +347,39 @@ def test_unknown_channels_do_not_crash(tmp_path):
         assert package_is_installed(prefix, "test-package")
         conda_inprocess("install", prefix, "zlib")
         assert package_is_installed(prefix, "zlib")
+
+
+@pytest.mark.skipif(not on_linux, reason="Only run on Linux")
+def test_use_cache_works_offline_fresh_install_keep(tmp_path):
+    """
+    https://github.com/conda/conda-libmamba-solver/issues/396
+
+    constructor installers have a `-k` switch (keep) to leave the
+    pkgs/ cache prepopulated. Offline updating from the cache should be a
+    harmless no-op, not a hard crash.
+    """
+    miniforge_url = (
+        "https://github.com/conda-forge/miniforge/releases/"
+        f"latest/download/Miniforge3-Linux-{os.uname().machine}.sh"
+    )
+    urlretrieve(miniforge_url, tmp_path / "miniforge.sh")
+    # bkfp: batch, keep, force, prefix
+    check_call(["bash", str(tmp_path / "miniforge.sh"), "-bkfp", tmp_path / "miniforge"])
+    env = os.environ.copy()
+    env["CONDA_ROOT_PREFIX"] = str(tmp_path / "miniforge")
+    env["CONDA_PKGS_DIRS"] = str(tmp_path / "miniforge" / "pkgs")
+    env["CONDA_ENVS_DIRS"] = str(tmp_path / "miniforge" / "envs")
+    env["HOME"] = str(tmp_path)  # ignore ~/.condarc
+    args = (
+        "update",
+        "-p",
+        tmp_path / "miniforge",
+        "--all",
+        "--dry-run",
+        "--override-channels",
+        "--channel=conda-forge",
+    )
+    kwargs = dict(capture_output=False, check=True, env=env)
```

Comment:
Calling dict without a good reason is a code-smell

```suggestion
    kwargs = {
        "capture_output": False,
        "check": True,
        "env": env,
    }
```